config/routes.rb

```diff
@@ -3,4 +3,8 @@
 
   # Defines the root path route ("/")
   # root "articles#index"
+  root 'users#index' # All users page
+  get '/users/:id', to: 'users#show', as: 'user' # Single user page
+  get '/users/:user_id/posts', to: 'posts#index_by_user', as: 'user_posts' # All posts by a given user page
+  get '/posts/:id', to: 'posts#show', as: 'post' # Single post page
```

Comment:
- [ ] I suggest that you use Resource routing because allows you to quickly declare all of the common routes for a given resourceful controller
```
 resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
```